File: DAL/MODELS.ProcureAccess/Entities/Dto/CriterionDto.cs
namespace MODELS.ProcureAccess.Entities.Dto;

public class CriterionDto : BaseDto
{
    public int Id { get; set; }
    public string? Name { get; set; }
    public string? Description { get; set; }
    public int? CriteriaFilterId { get; set; }
    public bool? IsDeleted { get; set; }
}

File: CLIENT/CLIENT.ProcureAccess/src/app/features/criteria/models/criterion.dto.ts
import 'reflect-metadata';
import { jsonObject, jsonMember, TypedJSON } from 'typedjson';

@jsonObject
export class CriterionDto {
  @jsonMember
  id: number;

  @jsonMember
  name: string;

  @jsonMember
  description: string;

  @jsonMember
  criteriaFilterId: number;

  constructor(
    pId: number, 
    pName: string, 
    pDescription: string, 
    pCriteriaFilterId: number) {
    this.id = pId;
    this.name = pName;
    this.description = pDescription;
    this.criteriaFilterId = pCriteriaFilterId;
  }
}
